src/middlewares/validator.js

```diff
@@ -1,13 +1,44 @@
 const createError = require('http-errors')
 const Ajv = require('ajv')
 const ajvKeywords = require('ajv-keywords')
-const {deepEqual} = require('assert')
+const ajvLocalize = require('ajv-i18n')
+const { deepEqual } = require('assert')
 
 let ajv
 let previousConstructorOptions
-const defaults = {v5: true, $data: true, allErrors: true}
+const defaults = {
+  v5: true,
+  coerceTypes: 'array', // important for query string params
```

Comment:
Hmm, isn't this a breaking change? `coerceTypes` is `false` by default

Reply:
From a code standpoint, yes. From an end-user standpoint, no. Since this middleware is paired with a json schema, having `coerceTypes ` enabled would make requests more forgiving when a `body` is passed in and enable the ability to parse query strings making them more usable. There are no cases where existing code would fail where they once passed. For example; if someone is using the current defaults their schema would have `"type":"string"` for all their query string params, in this case, no coercion would happen. `coerceTypes` is only applied if the types don't match. I added this option to support the use case when you expect an array from the query string, but only one item was passed in.

PS I wrote the `coerceTypes: 'array'` feature specifically for using ajv in an API middleware like we are here.

Reply:
> For example; if someone is using the current defaults their schema would have "type":"string" for all their query string params, in this case, no coercion would happen. 

Totally makes sense for me. And thanks for adding this, I see how it will be useful in my code already!

Reply:
@vladgolubev Along these lines, you may also be interested in this PR: https://github.com/epoberezkin/ajv-keywords/pull/64
I just had a new keyword added, `transform`, that allows to you to do some sanitization on strings being passed in. We're drafting up next steps for it here: https://github.com/epoberezkin/ajv-keywords/issues/66